src/guides/v2.3/howdoi/checkout/checkout_payment_new_field.md

```diff
@@ -0,0 +1,349 @@
+        <observer name="paymentfields_paymentfields_observer_frontend_sales_orderpaymentsavebefore" instance="Learning\CustomField\Observer\Frontend\Sales\OrderPaymentSaveBefore" />
+    </event>
+</config>
+```
+
+Then create the `app/code/Learning/CustomField/Observer/Frontend/Sales/OrderPaymentSaveBefore.php` file.
+
+{% collapsible Show code %}
+
+```php?start_inline=1
+<?php
+
+namespace Learning\CustomField\Observer\Frontend\Sales;
+
+use Magento\Framework\Event\Observer as EventObserver;
+use Magento\Framework\Event\ObserverInterface;
+use Magento\OfflinePayments\Model\Purchaseorder;
+use Magento\Framework\App\Request\DataPersistorInterface;
+
+class OrderPaymentSaveBefore implements \Magento\Framework\Event\ObserverInterface
+{
+    protected $order;
+    protected $logger;
+    protected $_serialize;
+    protected $quoteRepository;
+
+    public function __construct(
+        \Magento\Sales\Api\Data\OrderInterface $order,
+        \Magento\Quote\Api\CartRepositoryInterface $quoteRepository,
+        \Psr\Log\LoggerInterface $logger,
+        \Magento\Framework\Serialize\Serializer\Serialize $serialize
+    ) {
+        $this->order = $order;
+        $this->quoteRepository = $quoteRepository;
+        $this->logger = $logger;
+        $this->_serialize = $serialize;
+    }
+    /**
+     * Execute observer
+     *
+     * @param \Magento\Framework\Event\Observer $observer
+     * @return void
+     */
+    public function execute(\Magento\Framework\Event\Observer $observer)
+    {
+        $orderids = $observer->getEvent()->getOrderIds();
+        if(!$orderids){
+            foreach ($orderids as $orderid) {
+                $order = $this->_order->load($orderid);
+                $method = $order->getPayment()->getMethod();
+                if($method == 'purchaseorder') {
+                    $quote_id = $order->getQuoteId();
+                    $quote = $this->quoteRepository->get($quote_id);
+                    $paymentQuote = $quote->getPayment();
+                    $paymentOrder = $order->getPayment();
+                    $paymentOrder->setData('paymentpocomment',$paymentQuote->getPaymentpocomment());
+                    $paymentOrder->save();
+                }
+            }
+        }
+    }
+}
+```
+{% endcollapsible %}
+
+## Step 6: Verify that the module works {#verify-implementation}
+
+-  Run the below command to enable the new module
+
+```bash
+bin/magento module:enable Learning_CustomField
+```
+
+-  Run the below command to install the new module
+
+```bash
+bin/magento setup:upgrade
+```
+
+-  Run the below command to compile the code
+
+```bash
+bin/magento setup:di:compile
+```
+
+-  Run the below command to deploy the static files
+
+```bash
+bin/magento setup:static-content:deploy
+```
+
+Go to the storefront as a guest user and add the product to the cart. Go to the checkout page and select the **Purchase Order** payment. Verify that the **Purchase Order Comment** field is visible.
```

Comment:
Each sentence should be its own numbered step.

Reply:
@keharper Updated.